pkg/hostpath/controllerserver.go

```diff
@@ -119,7 +156,31 @@ func (hpc *hostPathController) CreateVolume(ctx context.Context, req *csi.Create
 			return nil, fmt.Errorf("failed to create volume %v: %w", req.GetName(), err)
 		}
 		klog.V(4).Infof("created volume %s at path %s", req.GetName(), filepath.Join(hpc.cfg.StoragePoolDataDir[storagePoolName], req.GetName()))
-	}	
+	}
+
+	if req.GetVolumeContentSource() != nil {
```

Comment:
If you didn't mean to call it twice, `if source := req.GetVolumeContentSource(); source != nil {` could work here
(I've seen two or more calls to this from a single func throughout the PR so might be missing something)